### ISSUE TITLE
Sweep inconsistant startup

### ISSUE THREAD

**Issue by planecrasher** [scanse/sweep-arduino#3]
This is an update of a report submitted through the scanse.io web page. 

I intend to modify the Arduino example code so that sensor readings are sent to processing code on my PC.  I want to explore mapping using the Sweep.

I .started work toward this with my Sweep (kickstarter reward) and an Arduino Mega and your example sketch. The Sweep had the latest firmware update (1.4).  The Sweep is powered by a separate 5v bench supply.  I think the Sweep works continuously with the Visualizer, but I'm not sure.  I've only used the Visualizer once or twice.  I am familiar with Arduino programming.

Operations seem intermittent. On most full powerup startups with the example code, motor speed and sampling info is reported to the monitor as the defaults (5/500) and 3 scan outputs are produced. If the program is then restarted, motor speed and sampling info is reported as -1/-1 and an error halt occurs. If I cycle the Sweep's 5v power, then restart the program, it will work the first time the program is started, indicated by the status LED solid blue, and default setting (5/500) reported and the sample output printed. After the initial program run the error halt occurs every time I restart the program.

What is happening when the device is spinning and the LED is solid blue, but the verify query in the example reports the speed and sampling as -1/-1?

Do you have any suggestions as to what I should do to get consistent operation every program restart?  Should I send a reset command on startup?  I could add that command to my library.  Or maybe lengthen time delays waiting for command completion? 

John Kleber

**Comment by planecrasher** [comment]
I have found that sending a reset command before starting the data collection works to make the data collection consistent without needing to cycle the Sweep power.  I've added a reset command to my version of the arduino library for Sweep.   All other library command seem to be working properly.

**Comment by dcyoung** [comment]
Hmmm, the `-1` return value for motor speed/sample rate could indicates that the proper response is not being parsed. Perhaps this is because the buffer still contains something from a previous cycle, and a flush would fix it. I'll investigate this.

Glad to hear the reset worked for you. The reset command should probably receive a dedicated method exposed in the public API. I'll create a new issue specifically for this, and hopefully push an update soon.

> I intend to modify the Arduino example code so that sensor readings are sent to processing code on my PC. 

If you are trying to get sweep data to a computer, the [sweep-sdk](https://github.com/scanse/sweep-sdk) is a far more more reliable and simple way to accomplish this. 

> I want to explore mapping using the Sweep.

That is a very interesting and exciting application for the Sweep. When you make progress, please share it on the [community forum](http://community.scanse.io/)! Though it is worth noting that mapping can be very computationally intensive. The arduino is far too under-powered to accomplish this locally. While it might be possible to use the arduino as an intermediate sending data to a more powerful computer, this would bottle-neck performance and create an unnecessary point of failure. It is probably easier to just use the [sweep-sdk](https://github.com/scanse/sweep-sdk) directly, as mentioned previously. However, if your application does require that an arduino serve as an intermediate, we'd love to hear more about it! Please feel free to share that on the community forum as well.
